File: 2696-minimum-string-length-after-removing-substrings/2696-minimum-string-length-after-removing-substrings.cpp
class Solution {
public:
    int minLength(string s) {
        vector<int> stack;
        for (char chr : s) {
            if (!stack.empty() && chr == 'B' && stack.back() == 'A') {
                stack.pop_back();
            } else if (!stack.empty() && chr == 'D' && stack.back() == 'C') {
                stack.pop_back();
            } else {
                stack.push_back(chr);
            }
        }

        return stack.size();
    }
};

File: 2696-minimum-string-length-after-removing-substrings/2696-minimum-string-length-after-removing-substrings.rb
# @param {String} s
# @return {Integer}
def min_length(s)
  stack = []
  s.each_char do |chr|
    if chr == "B" && stack.last == "A"
      stack.pop
    elsif chr == "D" && stack.last == "C"
      stack.pop
    else
      stack.push(chr)
    end
  end
  stack.size
end

